src/embind/embind_ts.js

```diff
@@ -271,8 +273,13 @@ var LibraryEmbind = {
         argStart = 2;
       }
       const args = [];
+      let x = 0;
       for (let i = argStart; i < argTypes.length; i++) {
-        args.push(new Argument(`_${i - argStart}`, argTypes[i]));
+        if (argsName.length && x < argsName.length) {
```

Comment:
Let's add an assert that if argNames are used it matches the expected number of arguments. This will help prevent bindings getting out of sync with their TS definitions.